plugins/woocommerce-admin/client/products/sections/product-details-section.tsx

```diff
@@ -141,6 +149,14 @@ export const ProductDetailsSection: React.FC = () => {
 							}
 						/>
 					) }
+					<RichTextEditor
+						label={ __( 'Description', 'woocommerce' ) }
+						blocks={ descriptionBlocks }
```

Comment:
If the `parse` function is not heavy in terms of performance then this `descriptionBlocks` state is not required and we can use only the internal `description` from the form context.
```javascript
    const descriptionControlProps = getInputProps( 'description' );
    ...
    blocks={ parse( descriptionControlProps?.value ?? '' ) }
    onChange={ ( blocks ) => {
        ...
        descriptionControlProps?.onChange( serialize( blocks ) );
    } }
    onBlur={ () => {
        ...
        descriptionControlProps?.onBlur();
    } }
```

Reply:
I'm not 100% sure how performant the `parse` function is.  Even as the component is right now, I have some concerns about its performance, but am trying not to prematurely optimize this method.

That said, we can try this to see if there are any issues.  We will probably need to memoize this portion `parse( descriptionControlProps?.value` otherwise blocks will be parsed on every rerender of this component, which may have non-insignificant performance impacts.

Reply:
I attempted to implement this, but because block changes are debounced 200ms for performance and we would parse the blocks on the next iteration, this will cause unintended side effects and force deselection of blocks.

<img width="368" alt="Screen Shot 2022-10-10 at 9 49 36 AM" src="https://user-images.githubusercontent.com/10561050/194917504-3fb46f1b-1fa3-4b77-bcb5-89122a1168fc.png">

```js
const descriptionBlocks = useMemo( () => {
    return parse( values.description || '' );
}, [ values.description ] );

<RichTextEditor
    label={ __( 'Description', 'woocommerce' ) }
    blocks={ descriptionBlocks }
    onChange={ ( blocks ) => {
        setValue( 'description', serialize( blocks ) );
    } }
/>
```

While I don't disagree that it would be nice to keep a single source of truth in the `description` value of the form, I think this may lead to issues with performance and quirks around block selection that we see here.